dbms/src/Storages/Transaction/tests/gtest_kvstore.cpp

```diff
@@ -36,47 +42,102 @@ extern void ChangeRegionStateRange(RegionState & region_state, bool source_at_le
 
 namespace tests
 {
+
+// TODO: Use another way to workaround calling the private methods on KVStore
 class RegionKVStoreTest : public ::testing::Test
 {
 public:
     RegionKVStoreTest() = default;
 
-    static void SetUpTestCase() {}
+    static void SetUpTestCase()
+    {
+        test_path = TiFlashTestEnv::getTemporaryPath("/region_kvs_test");
+    }
+
+    void SetUp() override
+    {
+        // clean data and create path pool instance
+        path_pool = createCleanPathPool(test_path);
+
+        reloadKVSFromDisk();
+
+        proxy_instance = std::make_unique<MockRaftStoreProxy>();
+        proxy_helper = std::make_unique<TiFlashRaftProxyHelper>(MockRaftStoreProxy::SetRaftStoreProxyFFIHelper(
+            RaftStoreProxyPtr{proxy_instance.get()}));
+        proxy_instance->init(100);
+
+        kvstore->restore(*path_pool, proxy_helper.get());
+    }
+
+    void TearDown() override
+    {
+        proxy_helper.reset();
+        proxy_instance.reset();
+        kvstore.reset();
+        path_pool.reset();
+    }
+
     static void testBasic();
     static void testKVStore();
     static void testRegion();
     static void testReadIndex();
     static void testNewProxy();
+    static void testKVStoreRestore();
+
+protected:
+    static KVStore & getKVS() { return *kvstore; }
+    static KVStore & reloadKVSFromDisk()
+    {
+        kvstore.reset();
+        auto & global_ctx = TiFlashTestEnv::getGlobalContext();
+        kvstore = std::make_unique<KVStore>(global_ctx, TiDB::SnapshotApplyMethod::DTFile_Directory);
+        // TODO: Do we need to recreate proxy_instance and proxy_helper here?
```

Comment:
MockRaftStoreProxy do not persists its own raftstate for now, so we can't  recreate proxy for now.

Reply:
updated